cmd/server/main.go

```diff
@@ -0,0 +1,50 @@
+package main
+
+import (
+	"flag"
+	"fmt"
+	"log"
+	"net"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"google.golang.org/grpc"
+
+	"github.com/Pratham700/github-search-service/internal/server"
+	pb "github.com/Pratham700/github-search-service/proto/proto"
+)
+
+func main() {
+	// Set the port for the gRPC server
+	port := flag.Int("port", 50051, "The port number to listen on")
```

Comment:
This is not a requirement, but the `Var` variant of flags helps avoid the use of a pointer here and generally makes the code using the flag easier to work read and write.

```suggestion
	var port int

	// Set the port for the gRPC server
	flag.IntVar(&port, "port", 50051, "The port number to listen on")
```